### PR TITLE
feat(codegen): allow generating tests (#7049)

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/server/supplements/recorder/javascript.ts [hunk 1]
```diff
@@ -84,8 +93,12 @@ export class JavaScriptLanguageGenerator implements LanguageGenerator {
 
     if (emitPromiseAll)
```
Comment (aslushnikov) [inline src/server/supplements/recorder/javascript.ts:94]: nit: braces

File: src/server/supplements/recorder/javascript.ts [hunk 2]
```diff
@@ -22,9 +22,16 @@ import { MouseClickOptions, toModifiers } from './utils';
...
 export class JavaScriptLanguageGenerator implements LanguageGenerator {
-  id = 'javascript';
-  fileName = '<javascript>';
+  id: string;
+  fileName: string;
   highlighter = 'javascript';
+  private _isTest: boolean;
+
+  constructor(isTest: boolean) {
+    this.id = isTest ? 'test' : 'javascript';
```
Comment (aslushnikov) [inline src/server/supplements/recorder/javascript.ts:31]:
```suggestion
    this.id = isTest ? 'playwright-test' : 'javascript';
```

to not clash with nunit in future

File: src/cli/cli.ts [hunk 3]
```diff
@@ -86,7 +86,7 @@ commandWithOpenOptions('open [url]', 'open page in browser specified via -b, --b
 commandWithOpenOptions('codegen [url]', 'open page and generate code for user actions',
     [
       ['-o, --output <file name>', 'saves the generated script to a file'],
-      ['--target <language>', `language to use, one of javascript, python, python-async, csharp`, language()],
+      ['--target <language>', `language to generate, one of javascript, test, python, python-async, csharp`, language()],
```
Comment (aslushnikov) [inline src/cli/cli.ts:89]: playwright-test

File: src/server/supplements/recorder/javascript.ts [hunk 4]
```diff
@@ -33,6 +40,8 @@ export class JavaScriptLanguageGenerator implements LanguageGenerator {
     formatter.add('// ' + actionTitle(action));
 
     if (action.name === 'openPage') {
+      if (this._isTest)
```
Comment (aslushnikov) [inline src/server/supplements/recorder/javascript.ts:43]: why ignoring this? It looks like it'll work just fine with context

File: src/server/supplements/recorder/javascript.ts [hunk 5]
```diff
@@ -131,6 +144,32 @@ export class JavaScriptLanguageGenerator implements LanguageGenerator {
...
+    return this.generateStandaloneHeader(options);
+  }
+
+  generateFooter(saveStorage: string | undefined): string {
+    if (this._isTest)
+      return this.generateTestFooter(saveStorage);
+    return this.generateStandaloneFooter(saveStorage);
+  }
+
+  generateTestHeader(options: LanguageGeneratorOptions): string {
```
Comment (aslushnikov) [inline src/server/supplements/recorder/javascript.ts:158]: nit: isn't it a private method?

**Review by aslushnikov — APPROVED** [review]

**Review by mxschmitt — APPROVED** [review]

**Pull request closed without merging**